intake_bluesky/__init__.py

```diff
@@ -1,4 +1,6 @@
-
+# Import intake to run driver discovery first and avoid circular import issues.
+import intake
```

Comment:
Is this import-then-delete necessary in general when using the new discovery mechanism? Seems a bit unfortunate.

Reply:
Intake does discovery at import time in its top-level `__init__.py`. Both the old and new discovery mechanism call `import_module` on each discovered module at that point. If that module in turn imports `intake`, as it often does, we have a circular import.

Just to be sure this part is clear: the `del intake` is just there to keep the namespace clean; it is not needed.

I'm not sure why the old approach _didn't_ need this, and the system seems fundamentally circular. Any idea, @martindurant?

Reply:
I'll dig into this further next week. It would be good understand fully what is happening before we update tens of repos to the new approach. It may be that we can avoid the necessity for this import by adjusting the discovery code. Thanks for the nudge, @ian-r-rose.